core/coreapi/interface/coreapi.go

```diff
@@ -32,9 +33,18 @@ type CoreAPI interface {
 	Object() ObjectAPI
 
 	// ResolvePath resolves the path using Unixfs resolver
-	ResolvePath(context.Context, Path) (Path, error)
+	ResolvePath(context.Context, Path) (ResolvedPath, error)
 
 	// ResolveNode resolves the path (if not resolved already) using Unixfs
 	// resolver, gets and returns the resolved Node
 	ResolveNode(context.Context, Path) (ipld.Node, error)
+
+	// ParsePath parses string path to a Path
+	ParsePath(string) (Path, error)
```

Comment:
I'm not really a fan of attaching helper functions to instance objects. What's the motivation?

Reply:
> What's the motivation?

We need `ResolvePath` on the interface as it is quite implementation specific, I thrown the other methods here for consistency.